plugins/http_plugin/include/eosio/http_plugin/common.hpp

```diff
@@ -0,0 +1,278 @@
+    using std::map;
+    using std::set;
+
+    namespace beast = boost::beast;           // from <boost/beast.hpp>
+    namespace http = boost::beast::http;      // from <boost/beast/http.hpp>
+    namespace asio = boost::asio;
+    namespace ssl = boost::asio::ssl;
+    using boost::asio::ip::tcp;               // from <boost/asio/ip/tcp.hpp>
+
+
+    namespace detail {
+        /**
+         * virtualized wrapper for the various underlying connection functions needed in req/resp processng
+        */
+        struct abstract_conn {
+            virtual ~abstract_conn() = default;
+            virtual bool verify_max_bytes_in_flight() = 0;
+            virtual bool verify_max_requests_in_flight() = 0;
+            virtual void handle_exception() = 0;
+
+            virtual void send_response(std::optional<std::string> body, int code) = 0;
+        };
+
+        using abstract_conn_ptr = std::shared_ptr<abstract_conn>;
+
+        /**
+         * internal url handler that contains more parameters than the handlers provided by external systems
+         */
+        using internal_url_handler = std::function<void(abstract_conn_ptr, string, string, url_response_callback)>;
+
+        /**
+         * Helper method to calculate the "in flight" size of a fc::variant
+         * This is an estimate based on fc::raw::pack if that process can be successfully executed
+         *
+         * @param v - the fc::variant
+         * @return in flight size of v
+         */
+        static size_t in_flight_sizeof( const fc::variant& v ) {
+           try {
+              return fc::raw::pack_size( v );
+           } catch( ... ) {}
+           return 0;
+        }
+
+        /**
+         * Helper method to calculate the "in flight" size of a std::optional<T>
+         * When the optional doesn't contain value, it will return the size of 0
+         *
+         * @param o - the std::optional<T> where T is typename
+         * @return in flight size of o
+         */
+        template<typename T>
+        static size_t in_flight_sizeof( const std::optional<T>& o ) {
+           if( o ) {
+              return in_flight_sizeof( *o );
+           }
+           return 0;
+        }
+
+        /**
+         * Helper method to calculate the "in flight" size of a string
+         * @param s - the string
+         * @return in flight size of s
+         */
+        static size_t in_flight_sizeof( const string& s ) {
+            return s.size();
+        }
+    }
+
+    // key -> priority, url_handler
+    typedef map<string,detail::internal_url_handler> url_handlers_type;
+
+    struct http_plugin_state {
+        string                         access_control_allow_origin;
+        string                         access_control_allow_headers;
+        string                         access_control_max_age;
+        bool                           access_control_allow_credentials = false;
+        size_t                         max_body_size{1024*1024};
```

Comment:
This was increased to `2*1024*1024` via https://github.com/eosnetworkfoundation/mandel/pull/160